src/Fixer/ClassNotation/MethodSeparationFixer.php

```diff
@@ -22,20 +26,30 @@
 use SplFileInfo;
 
 /**
- * Make sure there is one blank line above and below a method.
+ * Make sure there is one blank line above and below class elements.
  *
- * The exception is when a method is the first or last item in a 'classy'.
+ * The exception is when an element is the first or last item in a 'classy'.
  *
  * @author SpacePossum
  */
-final class MethodSeparationFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+final class MethodSeparationFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
 {
     /**
+     * @var array<string, true>
+     */
+    private $classElementTypes = [];
```

Comment:
do we need this property? can't we rely just on `$this->configuration['elements']` directly ?

Reply:
```
array(3) {
  [0]=>
  string(5) "const"
  [1]=>
  string(6) "method"
  [2]=>
  string(8) "property"
}
```

```
array(3) {
  ["const"]=>
  bool(true)
  ["method"]=>
  bool(true)
  ["property"]=>
  bool(true)
}
```

i.e. direct usage would be `in_array($type, $this->config['elements'], true)` vs. `isset($classElementTypes[$type])`, the last one is faster